src/config.cpp

```diff
@@ -68,11 +69,18 @@ struct ProviderOptionsArray_Element : JSON::Element {
       } else if (v.name == "dml") {
         v.name = "DML";
       }
+
+      if (std::find(providers_.begin(), providers_.end(), v.name) == providers_.end()) {
```

Comment:
Should we do this here? We could do it in the OnComplete() of the parent object and keep this object simple. Or could do it in the LoadConfig(). I don't think we can do it as late as CreateSessionOptionsFromConfig() as an empty providers list should mean to use cpu.

Reply:
Discussed offline that there are multiple places where this function is called. Doing it here will avoid needing to add the same code in multiple places. We can change this in the future where we avoid keeping the providers and the provider options in sync.